test/javascripts/page_test.coffee

```diff
@@ -85,3 +85,93 @@ describe 'Page', ->
         onlyKeys: ['a']
 
       @replaceStateStub.calledWith mockXHR.getResponseHeader('X-XHR-Redirected-To')
+
+  describe 'onReplace', ->
+
+    beforeEach ->
+      @fakebody = """
+        <div id='fakebody'>
+          <div id='foo'>
+            <div id='bar'>
+              <div id='foobar'></div>
+            </div>
+          </div>
+          <div id='baz'>
+            <div id='bat'></div>
+          </div>
+        </div>
+      """
+      $('body').append(@fakebody)
+
+
+    it 'calls the onReplace function only once', ->
+      node = $('#foo')[0]
+      refreshing = $("#fakebody")[0]
```

Comment:
I ask because if so we should let it handle the contains logic above, rather than putting in the browser-specific fixes ourselves